### PR TITLE
feat: adding clear, embeddings, insert, reverse, save, load

### PR DESCRIPTION
**Pull request by davidbp**

### CODE REVIEW

File: src/document_array_clear.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import pytest
+from jina import Document, DocumentArray
+from jina.types.arrays.memmap import DocumentArrayMemmap
+
+from .utils.benchmark import benchmark_time
+
+NUM_REPETITIONS = 10
+
+
+@pytest.mark.parametrize('num_docs', [100, 1000, 10_000])
```
Comment (nan-wang) [inline src/document_array_clear.py:10]: typo?
```suggestion
@pytest.mark.parametrize('num_docs', [100, 1000, 10000])
```

File: src/document_array_clear.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            iterations=NUM_REPETITIONS,
+            mean_time=mean_time,
+            std_time=std_time,
+            unit='ms',
+            metadata=dict(num_docs=num_docs),
+        )
+    )
+
+
+@pytest.mark.parametrize('num_docs', [100, 1000, 10_000])
```
Comment (nan-wang) [inline src/document_array_clear.py:37]:
```suggestion
@pytest.mark.parametrize('num_docs', [100, 1000, 10000])
```

File: src/document_array_embeddings.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+from jina.types.arrays.memmap import DocumentArrayMemmap
+
+from .utils.benchmark import benchmark_time
+
+NUM_REPETITIONS = 10
+
+# IMPORTANT: This benchmark currently is covered by
+# - document_array_save_binary_load_binary.py
+# - document_array_save_json_load_json.py
+# Only relevant if for future releases `.save` expands to other methods
```
Comment (nan-wang) [inline src/document_array_embeddings.py:13]: Is this part relevant to this file?

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: src/document_array_embeddings.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+
+@pytest.mark.parametrize('num_docs', [100, 1000, 10_000])
```
Comment (maateen) [inline src/document_array_embeddings.py:16]:
```suggestion
@pytest.mark.parametrize('num_docs', [100, 1000, 10000])
```

File: src/document_array_embeddings.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            iterations=NUM_REPETITIONS,
+            mean_time=mean_time,
+            std_time=std_time,
+            unit='ms',
+            metadata=dict(num_docs=num_docs),
+        )
+    )
+
+
+@pytest.mark.parametrize('num_docs', [100, 1000, 10_000])
```
Comment (maateen) [inline src/document_array_embeddings.py:46]:
```suggestion
@pytest.mark.parametrize('num_docs', [100, 1000, 10000])
```

File: src/document_array_insert.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import pytest
+from jina import Document, DocumentArray
+
+from .utils.benchmark import benchmark_time
+
+NUM_REPETITIONS = 10
+
+
+@pytest.mark.parametrize('num_docs', [100, 1000, 10_000])
```
Comment (maateen) [inline src/document_array_insert.py:9]:
```suggestion
@pytest.mark.parametrize('num_docs', [100, 1000, 10000])
```

File: src/document_array_reverse.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import pytest
+from jina import Document, DocumentArray
+
+from .utils.benchmark import benchmark_time
+
+NUM_REPETITIONS = 10
+
+
+@pytest.mark.parametrize('num_docs', [100, 1000, 10_000])
```
Comment (maateen) [inline src/document_array_reverse.py:9]:
```suggestion
@pytest.mark.parametrize('num_docs', [100, 1000, 10000])
```

File: src/document_array_save.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+NUM_REPETITIONS = 10
+
+# IMPORTANT: This benchmark currently is covered by
+# - document_array_save_binary_load_binary.py
+# - document_array_save_json_load_json.py
+# Only relevant if for future releases `.save` expands to other methods
+
+
+@pytest.mark.parametrize('num_docs', [100, 1000, 10_000])
```
Comment (maateen) [inline src/document_array_save.py:15]:
```suggestion
@pytest.mark.parametrize('num_docs', [100, 1000, 10000])
```

**Review by maateen — CHANGES_REQUESTED** [review]

File: src/document_array_clear.py (hunk 1 above)
Comment (davidbp) [reply src/document_array_clear.py:10]: Not a typo  https://www.python.org/dev/peps/pep-0515/ , I think it makes numbers with many zeros much more readable.

File: src/document_array_embeddings.py (hunk 3 above)
Comment (davidbp) [reply src/document_array_embeddings.py:13]: We can remove it, but probably will forget about it then
